app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -360,79 +402,79 @@ private void takeAPhoto(){
         startActivityForResult(intent, TAKE_PHOTO);
     }
 
-    private byte[] compress(Bitmap photo){
+    private byte[] compress(Bitmap photo) {
         int quality = 100;
 
-        if (photo.getByteCount()<= MAX_PHOTO_SIZE) {
+        if (photo.getByteCount() <= MAX_PHOTO_SIZE) {
             photo.compress(Bitmap.CompressFormat.PNG, quality, photoStream);
             return photoStream.toByteArray();
         }
 
         //
-        while(true) {
+        while (true) {
             photo.compress(Bitmap.CompressFormat.JPEG, quality, photoStream);
-            byte [] compressedPhoto = photoStream.toByteArray();
-            if(compressedPhoto.length <= MAX_PHOTO_SIZE || quality == 5){
+            byte[] compressedPhoto = photoStream.toByteArray();
+            if (compressedPhoto.length <= MAX_PHOTO_SIZE || quality == 5) {
                 return compressedPhoto;
-            }
-            else {
+            } else {
                 quality -= 5;
             }
         }
     }
 
-    protected void onActivityResult(int requestCode, int resultCode, Intent intent){
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
         Log.d("tag", "In Take Photo");
         TextView modePicture = (TextView) findViewById(R.id.modePhoto);
-        if (requestCode == TAKE_PHOTO){
+        if (requestCode == TAKE_PHOTO) {
 
-            if (resultCode == RESULT_OK){
+            if (resultCode == RESULT_OK) {
                 Log.d("tag", "Taking photo");
 
                 Bitmap photo = (Bitmap) intent.getExtras().get("data");
 
+                // Clear the old photoStream
+                photoStream.reset();
```

Comment:
This crucial segment was missing and as such, taking an image 2x led to the second's imageStream being APPENDED onto the first. This caused massive corruption within our database, and massive slowdowns of our app in general. The "Image 2x" bug has officially been squashed.